src/main/java/org/cristalise/restapi/ItemRoot.java

```diff
@@ -72,13 +71,14 @@
 import org.cristalise.kernel.utils.LocalObjectLoader;
 import org.cristalise.kernel.utils.Logger;
 import org.glassfish.jersey.media.multipart.FormDataParam;
-import org.json.JSONObject;
 import org.json.XML;
 
+import com.google.common.collect.ImmutableMap;
+
 @Path("/item/{uuid}")
 public class ItemRoot extends ItemUtils {
-
-    static Semaphore mutex = new Semaphore(1);
+	
+	private ScriptAccess scriptAccess = new ScriptAccess();

```

Comment:
Please create a ScriptUtils class and use  that in ItemRoot and ScriptAccess. I think it is not a good idea to instantiate a class with jaxb @Path annotation within a class which also contains different @Path declaration. These classes are instatiated by the jaxb routing and this kind of setup might have side effects.

Reply:
Pushed a commit for ScriptUtils